src/resty/aws/credentials/ProfileCredentials.lua

```diff
@@ -0,0 +1,41 @@
+--- EnvironmentCredentials class.
+-- @classmod EnvironmentCredentials
+
+
+-- Create class
+local Super = require "resty.aws.credentials.Credentials"
+local config = require "resty.aws.config"
+local ProfileCredentials = setmetatable({}, Super)
+ProfileCredentials.__index = ProfileCredentials
+
+
+--- Constructor, inherits from `Credentials`.
+--
+-- _Note_: this class will fetch the credentials upon instantiation. So it can be
+-- instantiated in the `init` phase where there is still access to the environment
+-- variables. The standard prefixes `AWS` and `AMAZON` are covered by the `config`
+-- module, so in case those are used, only the `config` module needs to be loaded
+-- in the `init` phase.
+-- @function aws:ProfileCredentials
+-- @param opt options table, additional fields to the `Credentials` class:
+function ProfileCredentials:new(opts)
```

Comment:
I suggest we rename it to `SharedCredentials` or `SharedFileCredentials`, to keep the same style with other AWS SDKs, like 
https://github.com/aws/aws-sdk-go/blob/main/aws/credentials/shared_credentials_provider.go
https://github.com/aws/aws-sdk-js/blob/307e82673b48577fce4389e4ce03f95064e8fe0d/lib/credentials/shared_ini_file_credentials.js